src/test/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -990,6 +971,23 @@ public synchronized void afterTest() throws IOException {
         randomlyResetClients(); /* reset all clients - each test gets its own client based on the Random instance created above. */
     }
 
+    @Override
+    public void beforeIndexDeletion() {
+        assertShardIndexCounter();
+    }
+
+    private void assertShardIndexCounter() {
+        final Collection<NodeAndClient> nodesAndClients = nodes.values();
+        for (NodeAndClient nodeAndClient : nodesAndClients) {
+            IndicesService indexServices = getInstance(IndicesService.class, nodeAndClient.name);
+            for (IndexService indexService : indexServices) {
+                for (IndexShard indexShard : indexService) {
+                    assertThat(indexShard.getOperationsCount(), equalTo(1));
```

Comment:
what if this shard got relocated etc. then the counter is `0` for a short amount of time. I think we need to check this and if the counter is `0` here the shard must be closed so maybe we can check this too?